functions/feedback/v1.js

```diff
@@ -0,0 +1,248 @@
+'use strict'
+
+const azureStorageHelper = require('../lib/azure-storage-helper')
+const azureQueueService = azureStorageHelper.getPromisifiedAzureQueueService()
+const azureTableService = azureStorageHelper.getPromisifiedAzureTableService()
+const moment = require('moment')
+const R = require('ramda')
+const sqlService = require('../lib/sql/sql.service')
+const uuid = require('uuid/v4')
+
+const feedbackQueueName = 'pupil-feedback'
+const pupilEventTableName = 'pupilEvent'
+const functionName = 'feedback'
+let logger
+
+/**
+ * Retrieve feedback messages from the queue, 32 at a time
+ * @return {Promise<null|*>}
+ */
+async function fetchFeedBackMessages () {
+  try {
+    return await azureQueueService.getMessagesAsync(feedbackQueueName, { numOfMessages: 32, visibilityTimeout: 60 })
+  } catch (error) {
+    logger.error('Failed to fetch new pupil-feedback messages from the queue: ' + error.message)
+    logger.error(error)
+    // If we are having difficulty getting new messages from the queue, bail out and let the next timer trigger pick up
+    // the remaining work
+    return null
+  }
+}
+
+/**
+ * Generate objects suitable for inserting into the pupilData table
+ * @param messages - array of messages
+ * @return {Object[]}
+ */
+function pupilEventData (messages) {
+  return messages.map(msg => {
+    return {
+      PartitionKey: msg.message.checkCode,
+      RowKey: uuid(),
+      eventType: 'feedback',
+      payload: JSON.stringify(msg.message),
+      processedAt: new Date()
+    }
+  })
+}
+
+/**
+ * Insert array of entities into table storage
+ * @param {Object[]} tableData - array of entities
+ * @param {String} tableName - name of the azure storage table
+ * @return {Promise<*>}
+ */
+async function parallelInsertToAzureStorageTable (tableData, tableName) {
+  const insertPromises = tableData.map(entity => {
+    azureTableService.insertEntityAsync(tableName, entity)
```

Comment:
This will process whatever the batchSize is set to - 32 at the moment or smaller